custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,221 @@
+            return
+
+        self.update_temp(new_state)
+        self.schedule_update_ha_state()
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_ON_OFF in self.device.states:
+            if (
+                self.executor.select_state(OverkizState.CORE_ON_OFF)
+                == OverkizCommandParam.OFF
+            ):
+                return HVAC_MODE_OFF
+
+        if (
+            OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE
+            in self.device.states
+        ):
+            return TAHOMA_TO_HVAC_MODES[
+                self.executor.select_state(
+                    OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE
+                )
+            ]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.executor.async_execute_command(
+            "setActiveMode", HVAC_MODES_TO_TAHOMA[hvac_mode]
+        )
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return TAHOMA_TO_PRESET_MODES[
+            self.executor.select_state(
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_SETPOINT_MODE
+            )
+        ]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self.executor.async_execute_command(
+            "setManuAndSetPointModes", PRESET_MODES_TO_TAHOMA[preset_mode]
+        )
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        current_operation = self.executor.select_state(
+            "ovp:HeatingTemperatureInterfaceOperatingModeState"
+        )
+
+        if current_operation in TAHOMA_TO_HVAC_ACTION:
+            return TAHOMA_TO_HVAC_ACTION[current_operation]
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            if self.preset_mode == PRESET_NONE:
+                return None
+            return self.executor.select_state(MAP_PRESET_TEMPERATURES[self.preset_mode])
+        return None
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._current_temperature
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new temperature."""
+
+        mode = self.executor.select_state(
```

Comment:
In theory you could here check as well for `self.preset_mode`, since that is using the `OVP_HEATING_TEMPERATURE_INTERFACE_SETPOINT_MODE` as well. This would mean that you don't have to pull the state again, but can use the already retrieved state.

@tetienne / @vlebourl what would be your preference?

Reply:
I think it can be removed.
There's no way to set the current temperature in auto mode, if it's in a manual mode, then you change current preset (eco / comfort / freeze).
That's why I've asked if it's possible to changes the feature dynamically to allow changing temperature only in manual mode.

Reply:
I've done some test, it works well with the SUPPORT_TARGET_TEMPERATURE, but  there's no way to set a derogated temperature in auto mode.

It affects the current mode (eco/comfort) and stay persisted if you change to manual mode. This is contrary to how the physical device work.

I think it should stay disabled, and should be done thought the NumberEntity instead to prevent surprise from the users.